src/identifier.rs

```diff
@@ -90,6 +91,14 @@ impl std::borrow::Borrow<str> for Identifier {
     }
 }
 
+impl FromStr for Identifier {
+    type Err = ErrorKind;
```

Comment:
I don't love using `ErrorKind` as an actual error, since it doesn't implement the `Error` trait.

Can we think of any reason why we would actually need this impl? We can just cut it for now, and keep `Name`...